packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,52 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    const attributeMap = {};
+    do {
+      // we are trying to parse a node which has the following syntax
+      // with type: "json"
+      // [with -> keyword], [type -> Identifier], [":" -> token for colon], ["json" -> StringLiteral]
+      const node = this.startNode();
+      node.key = this.parseIdentifier(true);
+
+      // check if we already have an entry for an attribute
+      // if a duplicate entry found, throw an error
+      if (attributeMap[node.key.name]) {
+        throw this.raise(
+          this.state.start,
+          Errors.ModuleAttributesWithDuplicateKeys,
+        );
```

Comment:
If you modify the error message like this:

```
Module attributes should not include duplicate keys ("%1")
```

then you can pass `node.key.name` as an additional parameter to `this.raise` and it will be included in the error message.